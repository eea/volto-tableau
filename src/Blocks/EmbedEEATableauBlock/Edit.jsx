import React from 'react';
import BlockDataForm from '@plone/volto/components/manage/Form/BlockDataForm';
import { SidebarPortal } from '@plone/volto/components';
import { getContent } from '@plone/volto/actions';
import View from './View';
import Schema from './schema';
import { connect } from 'react-redux';
import { compose } from 'redux';

const Edit = (props) => {
  const { data, block, onChangeBlock, id } = props;
  const schema = React.useMemo(() => Schema(props), [props]);

  React.useEffect(() => {
    if (!Object.hasOwn(data, 'show_sources')) {
      onChangeBlock(block, {
        ...data,
        show_sources: true,
      });
    }
  }, [block, data, onChangeBlock]);

  return (
    <>
      <View data={data} id={id} />
      <SidebarPortal selected={props.selected}>
        <BlockDataForm
          block={block}
          title={schema.title}
          schema={schema}
          onChangeField={(id, value) => {
            props.onChangeBlock(block, {
              ...data,
              [id]: value,
            });
          }}
          formData={data}
        />
      </SidebarPortal>
    </>
  );
};

export default compose(
  connect(
    (state, props) => ({
      block_data: state.content.data,
      data_provenance:
        state.content.subrequests?.[props.id]?.data?.data_provenance,
    }),
    {
      getContent,
    },
  ),
)(Edit);
