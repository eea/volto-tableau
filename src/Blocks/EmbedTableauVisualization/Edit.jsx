import React from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { injectIntl } from 'react-intl';
import BlockDataForm from '@plone/volto/components/manage/Form/BlockDataForm';
import { SidebarPortal } from '@plone/volto/components';
import getSchema from './schema';

import View from './View';

const Edit = (props) => {
  const { block, data, selected, onChangeBlock, data_query } = props;
  const schema = React.useMemo(() => getSchema(props), [props]);

  React.useEffect(() => {
    if (data_query) {
      data.data_query_params = data_query;
    }
  }, [data_query, data]);

  return (
    <React.Fragment>
      <View mode="edit" {...props} />
      <SidebarPortal selected={selected}>
        <BlockDataForm
          block={block}
          schema={schema}
          title={schema.title}
          onChangeBlock={onChangeBlock}
          onChangeField={(id, value) => {
            onChangeBlock(block, {
              ...data,
              [id]: value,
            });
          }}
          formData={data}
        />
      </SidebarPortal>
    </React.Fragment>
  );
};

export default compose(
  connect((state, props) => ({
    data_query: state.content?.data?.data_query,
  })),
  injectIntl,
)(Edit);
