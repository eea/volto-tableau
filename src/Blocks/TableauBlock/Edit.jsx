import React from 'react';
import { SidebarPortal } from '@plone/volto/components';
import BlockDataForm from '@plone/volto/components/manage/Form/InlineForm';
import config from '@plone/volto/registry';
import getSchema from './schema';
import View from './View';

const Edit = (props) => {
  const viz = React.useRef();
  const [vizState, setVizState] = React.useState({
    loaded: false,
    loading: false,
    error: null,
  });

  const schema = React.useMemo(() => getSchema(config, viz.current, vizState), [
    vizState,
  ]);

  return (
    <>
      <View {...props} mode="edit" setVizState={setVizState} ref={viz} />
      <SidebarPortal selected={props.selected}>
        <BlockDataForm
          block={props.block}
          schema={schema}
          title={schema.title}
          onChangeField={(id, value) => {
            props.onChangeBlock(props.block, {
              ...props.data,
              [id]: value,
            });
          }}
          formData={props.data}
        />
      </SidebarPortal>
    </>
  );
};

export default Edit;
