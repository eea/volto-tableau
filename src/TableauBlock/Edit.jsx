import React from 'react';
import { SidebarPortal } from '@plone/volto/components';
import InlineForm from '@plone/volto/components/manage/Form/InlineForm';
import config from '@plone/volto/registry';
import getSchema from './schema';
import View from './View';

const Edit = (props) => {
  const [schema] = React.useState(getSchema(config));

  return (
    <>
      <View {...props} mode="edit" />

      <SidebarPortal selected={props.selected}>
        <InlineForm
          schema={schema}
          title={schema.title}
          onChangeField={(id, value) => {
            props.onChangeBlock(props.block, {
              ...props.data,
              [id]: value,
            });
          }}
          formData={props.data}
        />
      </SidebarPortal>
    </>
  );
};

export default Edit;
