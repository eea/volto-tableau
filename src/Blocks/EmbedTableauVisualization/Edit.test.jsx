import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-intl-redux';
import config from '@plone/volto/registry';

import Edit from './Edit';
import installEmbedTableau from '.';

installEmbedTableau(config);

describe('Edit', () => {
  const data = {
    '@type': 'embed_tableau_visualization',
    dataprotection: {
      enabled: false,
    },
    tableau_vis_url: 'http://localhost:3000/tableau-ct',
    with_download: true,
    with_more_info: true,
    with_notes: true,
    with_share: true,
  };

  it('should render the component', () => {
    const component = renderer.create(
      <Provider store={global.store}>
        <Edit
          id="my-tableau"
          data={data}
          pathname="/news"
          selected={false}
          block="1234"
          index={1}
          onChangeBlock={() => {}}
          onSelectBlock={() => {}}
          onDeleteBlock={() => {}}
          onFocusPreviousBlock={() => {}}
          onFocusNextBlock={() => {}}
          handleKeyDown={() => {}}
          content={{}}
          useVisibilitySensor={false}
        />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
