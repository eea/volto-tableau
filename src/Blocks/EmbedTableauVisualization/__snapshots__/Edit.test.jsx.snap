// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit should render the component 1`] = `
Array [
  <div
    className="embed-tableau"
  >
    <div
      className="privacy-protection-wrapper"
      style={
        Object {
          "height": "auto",
          "minHeight": "200px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        className="tableau-wrapper"
      >
        <div
          className="tableau tableau-2.8.0"
          style={
            Object {
              "height": "700px",
            }
          }
        />
      </div>
    </div>
  </div>,
  <div
    id="sidebar"
  >
    <div
      className="ui form"
    >
      <header
        className="header pulled"
      >
        <h2>
          Embed Dashboard (Tableau)
        </h2>
      </header>
      <div
        id="blockform-fieldset-default"
      >
        <div
          className="ui segment form attached"
        >
          <div
            className="mocked-url-widget"
            id="mocked-field-tableau_vis_url"
          >
            Tableau visualization
             - 
            No description
          </div>
          <div
            className="mocked-default-widget"
            id="mocked-field-tableau_height"
          >
            <a
              href="https://developer.mozilla.org/en-US/docs/Web/CSS/height"
              rel="noreferrer"
              target="_blank"
            >
              CSS height
            </a>
             - 
            Tableau height
          </div>
        </div>
      </div>
      <div
        className="accordion ui fluid styled form"
      >
        <div
          id="blockform-fieldset-toolbar"
        >
          <div
            className="active title"
            onClick={[Function]}
          >
            Toolbar
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": undefined,
                }
              }
              onClick={null}
              style={
                Object {
                  "fill": "currentColor",
                  "height": "20px",
                  "width": "auto",
                }
              }
              viewBox=""
              xmlns=""
            />
          </div>
          <div
            className="content active"
          >
            <div
              aria-hidden={false}
              className="rah-static rah-static--height-auto"
              style={
                Object {
                  "height": "auto",
                  "overflow": "visible",
                }
              }
            >
              <div
                style={
                  Object {
                    "WebkitTransition": "opacity 500ms ease 0ms",
                    "transition": "opacity 500ms ease 0ms",
                  }
                }
              >
                <div
                  className="ui segment attached"
                >
                  <div
                    className="mocked-boolean-widget"
                    id="mocked-field-with_notes"
                  >
                    Show note
                     - 
                    No description
                  </div>
                  <div
                    className="mocked-boolean-widget"
                    id="mocked-field-with_sources"
                  >
                    Show sources
                     - 
                    Will show sources set in this page Data provenance
                  </div>
                  <div
                    className="mocked-boolean-widget"
                    id="mocked-field-with_more_info"
                  >
                    Show more info
                     - 
                    No description
                  </div>
                  <div
                    className="mocked-boolean-widget"
                    id="mocked-field-with_download"
                  >
                    Show download button
                     - 
                    No description
                  </div>
                  <div
                    className="mocked-boolean-widget"
                    id="mocked-field-with_share"
                  >
                    Show share button
                     - 
                    No description
                  </div>
                  <div
                    className="mocked-boolean-widget"
                    id="mocked-field-with_enlarge"
                  >
                    Show enlarge button
                     - 
                    No description
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="accordion ui fluid styled form"
      >
        <div
          id="blockform-fieldset-privacy"
        >
          <div
            className="title"
            onClick={[Function]}
          >
            Privacy
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": undefined,
                }
              }
              onClick={null}
              style={
                Object {
                  "fill": "currentColor",
                  "height": "20px",
                  "width": "auto",
                }
              }
              viewBox=""
              xmlns=""
            />
          </div>
          <div
            className="content"
          >
            <div
              aria-hidden={true}
              className="rah-static rah-static--height-zero"
              style={
                Object {
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <div
                style={
                  Object {
                    "WebkitTransition": "opacity 500ms ease 0ms",
                    "opacity": 0,
                    "transition": "opacity 500ms ease 0ms",
                  }
                }
              >
                <div
                  className="ui segment attached"
                >
                  <div
                    className="mocked-default-widget"
                    id="mocked-field-dataprotection"
                  >
                    No title
                     - 
                    No description
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
